### PR TITLE
fix: added option to use application url for domain's cookies

### PR DESCRIPTION
**Pull request by drorrose**
@davidmeirlevy 
We didn't have a domain to test the changes on, so please do?

### CODE REVIEW

File: config/index.js [hunk 1]
```diff
@@ -5,6 +5,7 @@ const THIRTY_DAYS = 1000 * 60 * 60 * 24 * 30
 
 module.exports = {
 	mongoUri: process.env.MONGO_URI || "mongodb://localhost/auth-service",
```
Comment (davidmeirlevy) [inline config/index.js:7]: In this case - don't use a fallback value. Especially not my domain :).

File: server/services/tokens.js [hunk 2]
```diff
@@ -29,7 +29,10 @@ function getUniqueId (creationTime = Date.now().toString()) {
 }
 
 function setCookie (res, cookieId, maxAge = cookieTokenExpiration) {
-	res.cookie('token', cookieId, { maxAge, httpOnly: true })
+	const { applicationUrl } = require('../../config')
+	const finalUrl = '.' + new URL(applicationUrl).hostname
```
Comment (davidmeirlevy) [inline server/services/tokens.js:33]: The app url will probably be a full URL including protocol and "www".
They should be removed and changed programmatically (and once, not on each cookie set).

**Review by davidmeirlevy — CHANGES_REQUESTED** [review]

File: config/index.js (hunk 1 above)
Comment (drorrose) [reply config/index.js:7]: Done :)

File: server/services/tokens.js (hunk 2 above)
Comment (drorrose) [reply server/services/tokens.js]: @davidmeirlevy 
const finalUrl = '.' + new URL(applicationUrl).hostname
This line should do it, as the hostname will return only the domain itself. 

I definitely agree with the single handling for the URL settings, so I fixed that, and I added a dedicated function to get the needed cookie parameters. Hopefully, this should do the trick, but let me know if it doesn't.

File: server/services/tokens.js [hunk 3]
```diff
@@ -1,5 +1,11 @@
 const jwt = require('jsonwebtoken')
-const { jwtSecret, refreshTokenSecret, tokenExpiration, cookieTokenExpiration } = require('../../config')
+const { jwtSecret, refreshTokenSecret, tokenExpiration, 
+		cookieTokenExpiration, applicationUrl } = require('../../config')
+
+let finalDomainUrl = undefined
+if (applicationUrl) {
+	finalDomainUrl = '.' + new URL(applicationUrl).hostname
```
Comment (davidmeirlevy) [inline server/services/tokens.js:7]: I don't believe it's enough.
let's say the domain is: https://www.facebook.com

Will it be converted to ".facebook.com"? ("www" should be removed too).

**Review by davidmeirlevy — CHANGES_REQUESTED** [review]

File: server/services/tokens.js (hunk 3 above)
Comment (drorrose) [reply server/services/tokens.js]: You are right, the "www" is not removed. The last commit handles it!

**Review by davidmeirlevy — APPROVED** [review]

**Pull request merged**